src/liballoc/boxed.rs

```diff
@@ -201,6 +203,7 @@ impl<T: ?Sized> Box<T> {
     /// ```
     #[unstable(feature = "box_into_raw_non_null", issue = "47336")]
     #[inline]
+    #[rustc_deprecated(reason = "Use `.into()`", since = "1.32.0")]
```

Comment:
I would prefer not to deprecated this. Inherent associated functions are more discoverable and avoid needing to guide type inference, as you see in the `into_raw` implementation above.

```rust
Box::into_raw_non_null(b).as_ptr()
```

```rust
let p: NonNull<T> = b.into();
p.as_ptr()
```

Reply:
@SimonSapin suggested this.  I'm happy either way.

Do we want one implementation to delegate to the other?  And if yes, which way?

Reply:
@dtolnay Should every single impl of conversion traits have a corresponding inherent method? If not, what makes this one special?

Reply:
Perhaps the very fact that the From/Into traits are type conversions makes them special with regards to ergonomics and type inference.  My understanding was that was one of the reasons for having both From _and_ Into, i.e `Foo::from` avoids type inference, while `bar.into()` method-syntax is nicer to use.  Being forced by the coherence rules I think it might be worth keeping the associated function.